WordPress/Classes/ViewRelated/Views/StoppableProgressIndicatorView.swift

```diff
@@ -0,0 +1,117 @@
+import Foundation
+import UIKit
+
+private let rotationAnimationKey = "rotation"
+
+@objcMembers class StoppableProgressIndicatorView: UIView {
+
+    private let progressIndicator: CAShapeLayer
+    let stopButton: UIControl
+
+    var hidesWhenStopped: Bool = true {
+        didSet {
+            if hidesWhenStopped, !isAnimating {
+                isHidden = true
+            }
+        }
+    }
+
+    var mayStop: Bool {
+        get { !stopButton.isHidden }
+        set { stopButton.isHidden = !newValue }
+    }
+
+    var isAnimating: Bool {
+        progressIndicator.animation(forKey: rotationAnimationKey) != nil
+    }
+
+    private var resumeAnimation: Bool = true
+
+    override init(frame: CGRect) {
+        stopButton = UIControl(frame: .zero)
+        progressIndicator = CAShapeLayer()
+        progressIndicator.isHidden = true
+        progressIndicator.lineWidth = 2
+        super.init(frame: frame)
+
+        layer.addSublayer(progressIndicator)
+        addSubview(stopButton)
+        stopButton.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            stopButton.centerXAnchor.constraint(equalTo: centerXAnchor),
+            stopButton.centerYAnchor.constraint(equalTo: centerYAnchor),
+            stopButton.widthAnchor.constraint(equalTo: stopButton.heightAnchor),
+            stopButton.widthAnchor.constraint(equalTo: widthAnchor, multiplier: 0.3),
+        ])
+
+        updateAppearance()
+        NotificationCenter.default.addObserver(self, selector: #selector(enterForeground), name: UIApplication.willEnterForegroundNotification, object: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func tintColorDidChange() {
+        super.tintColorDidChange()
+        updateAppearance()
+    }
```

Comment:
Looking at the videos in the PR description, the tint color of both implementations stands out as the default iOS color against the rest of the Jetpack app the brand green.

Sort of out of scope for this PR, given that the MRProgress implementation also had this inconsistency, but maybe in building this you might have noticed an appropriate spot in the codebase where to set the tint color?

Reply:
Yeah, I noticed that. I didn't look into it though. 😓